openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py

```diff
@@ -73,10 +73,26 @@ def _delete_mail_channel_partner_duplicate_records(env):
     )
 
 
+def _add_follwers_from_mail_channel(env):
+    # As mail_channel field is removed from mail.follower in v15
+    # so we need to add followers which present in that channel
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO mail_followers (res_model, res_id, partner_id)
+        SELECT mf.res_model, mf.res_id, mcp.partner_id from mail_followers mf
+        join mail_channel mc on mf.channel_id = mc.id
+        join mail_channel_partner mcp on mcp.channel_id = mc.id
+        ON CONFLICT (res_model, res_id, partner_id) DO NOTHING
```

Comment:
Em thêm check user đang hoạt động nữa nhé
join res_users ru on ru.partner_id = mcp.partner_id AND ru.active = true
cc @royleviindoo

Reply:
@daonamutc1 Dạ vâng ạ cc @royleviindoo